frontend/app/mocks/subscription-api.server.ts

```diff
@@ -29,6 +29,30 @@ export function getSubscriptionApiMockHandlers() {
   log.info('Initializing Subscription API mock handlers');
 
   return [
+    //
+    // Handler for GET request to retrieve user by userId
+    //
+    http.get('https://api.cdcp.example.com/api/v1/users', ({ request }) => {
+      log.debug('Handling request for [%s]', request.url);
+
+      const url = new URL(request.url);
+      const raoidcUserId = url.searchParams.get('raoidcUserId');
+
+      const parsedRaoidUserId = z.string().safeParse(raoidcUserId);
+
+      if (!parsedRaoidUserId.success) {
+        throw new HttpResponse(null, { status: 400 });
+      }
+
+      const userEntity = db.user.findFirst({
+        where: { userAttributes: { value: { equals: parsedRaoidUserId.data } } },
+      });
+
+      return HttpResponse.json({
+        userEntity,
+      });
```

Comment:
Since this is a filtering request over a collection, the response from the API will be a collection with 0 or more results.

The JSON must be of the form:

``` json
{
  "_embedded": {
    "users": [
      // 0 or more user resources
    ]
  }
}
```

The response will also contain the HATEOAS links, but if those are not relevant then I don't think there's any reason to include them in the mock response.

Reply:
Also, each user returned in the response will have HATEOAS links to their sub-resources (ie: subscriptions), so make sure you account for that.

Reply:
Here is a copy/paste of an actual API response for a `user` resource:

``` json
{
  "id": "76c48130-e1d4-4c2f-8dd0-1c17f9bbb4f6",
  "email": "unverified@example.com",
  "emailVerified": false,
  "userAttributes": [
    {
      "name": "RAOIDC_USER_ID",
      "value": "b5336580-d93d-4da9-9e19-c2a5e098bd08"
    }
  ],
  "createdBy": "flyway-community-edition",
  "createdDate": "2024-05-31T10:41:20.788578Z",
  "lastModifiedBy": "flyway-community-edition",
  "lastModifiedDate": "2024-05-31T10:41:20.788578Z",
  "_links": {
    "self": {
      "href": "http://localhost:8080/api/v1/users/76c48130-e1d4-4c2f-8dd0-1c17f9bbb4f6"
    },
    "subscriptions": {
      "href": "http://localhost:8080/api/v1/users/76c48130-e1d4-4c2f-8dd0-1c17f9bbb4f6/subscriptions"
    }
  }
}
```

Reply:
fixed

Reply:
`users` is a collection.. ex:

``` javascript
{
  _embedded: {
    users: [userEntity]
  }
}
```

Reply:
fixed